cvat/apps/engine/training.py

```diff
@@ -0,0 +1,552 @@
+            'Authorization': f'bearer_token {self.token}',
+        }
+        response = self.request(method='GET', url=url, headers=headers)
+        return response
+
+    @retry()
+    def __get_project(self, project_id: str) -> dict:
+        url = f'{self.host}/v2/projects/{project_id}'
+        headers = {
+            'Authorization': f'bearer_token {self.token}',
+        }
+        response = self.request(method='GET', url=url, headers=headers)
+        return response
+
+    @retry()
+    def __get_server_status(self) -> dict:
+        url = f'{self.host}/v2/status'
+        headers = {
+            'Authorization': f'bearer_token {self.token}',
+        }
+        response = self.request(method='GET', url=url, headers=headers)
+        return response
+
+    @retry()
+    def __get_tasks(self) -> List[dict]:
+        url = f'{self.host}/v2/tasks'
+        headers = {
+            'Authorization': f'bearer_token {self.token}',
+        }
+        response = self.request(method='GET', url=url, headers=headers)
+        return response
+
+    def __delete_token(self):
+        cache.delete(self.token_key)
+
+    @retry()
+    def __upload_annotation(self, project_id: str, image_id: str, annotation: List[dict]):
+        url = f'{self.host}/v2/projects/{project_id}/media/images/{image_id}/annotations'
+        headers = {
+            'Authorization': f'bearer_token {self.token}',
+            'Content-Type': 'application/json'
+        }
+        data = {
+            'image_id': image_id,
+            'data': annotation
+        }
+        response = self.request(method='POST', url=url, headers=headers, json=data)
+        return response
+
+    @retry()
+    def __upload_image(self, project_id: str, buffer, image_path: str = '') -> dict:
+        url = f'{self.host}/v2/projects/{project_id}/media/images'
+        # print('file_path', image_path, os.path.isfile(image_path))
+        files = {'file': buffer}
+        # files = {'file': open(image_path, 'rb')}
+        headers = {
+            'Authorization': f'bearer_token {self.token}',
+        }
+        response = self.request(method='POST', url=url, headers=headers, files=files)
+        return response
+
+    @property
+    def project_id_key(self):
+        return f'{self.host}_{self.username}_project_id'
+
+    @property
+    def token(self) -> str:
+        def get_token(host: str, username: str, password: str) -> dict:
+            url = f'{host}/v2/authentication'
+            data = {
+                'username': (None, username),
+                'password': (None, password),
+            }
+            response = requests.post(url=url, files=data, verify=False)
+            return response.json()
+
+        try:
+            token = cache.get(self.token_key)
+        except CacheMiss:
+            response = get_token(self.host, self.username, self.password)
+            cache.set(cache_key=self.token_key, data=response['secure_token'], timeout=response['expires_in'])
```

Comment:
For a training server without authorization it doesn't work. It returns a dummy token with one key (wihtout expires_in).

```suggestion
            cache.set(cache_key=self.token_key, data=response['secure_token'], timeout=response.get('expires_in'))
```